From 1609e64e9615afc5d84925a2e0324c5850f52821 Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
Date: Thu, 12 Apr 2018 00:16:18 +0700
Subject: [PATCH 412/433] usb: host: xhci-plat: treat R8A774[2,4] with
 XHCI_NO_64BIT_SUPPORT quirk

As of commit f8cfd937b40b ("usb: host: xhci-plat: Fix cannot work if
R-Car Gen2/3 run on above 4GB phys"), also treat R8A7742 (RZ/G1H) and
R8A7744 (RZ/G1N) with the same xhci quirk as RCar Gen2/3 SoCs.

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 drivers/usb/host/xhci-plat.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/host/xhci-plat.c b/drivers/usb/host/xhci-plat.c
index c12aa9f..7b31368 100644
--- a/drivers/usb/host/xhci-plat.c
+++ b/drivers/usb/host/xhci-plat.c
@@ -54,7 +54,9 @@ static void xhci_plat_quirks(struct device *dev, struct xhci_hcd *xhci)
 	 * to call dma_set_coherent_mask(dev, DMA_BIT_MASK(32)) in
 	 * xhci_gen_setup().
 	 */
-	if (of_device_is_compatible(of_node, "renesas,xhci-r8a7743"))
+	if (of_device_is_compatible(of_node, "renesas,xhci-r8a7743") ||
+	    of_device_is_compatible(of_node, "renesas,xhci-r8a7744") ||
+	    of_device_is_compatible(of_node, "renesas,xhci-r8a7742"))
 		xhci->quirks |= XHCI_NO_64BIT_SUPPORT;
 }
 
-- 
2.7.4

